packages/lesswrong/server/scripts/mergeAccounts.js

```diff
@@ -0,0 +1,149 @@
+  await editMutation({
+    collection,
+    documentId,
+    set: {[fieldName]: targetUserId},
+    unset: {},
+    validate: false,
+  })
+}
+
+const transferCollection = async ({sourceUserId, targetUserId, collectionName, fieldName = "userId"}) => {
+  const collection = getCollection(collectionName)
+  const documents = await collection.find({[fieldName]: sourceUserId}).fetch()
+  // eslint-disable-next-line no-console
+  console.log(`Transferring ${documents.length} documents in collection ${collectionName}`)
+  for (const doc of documents) {
+    await transferOwnership({documentId: doc._id, targetUserId, collection, fieldName})
+    // Transfer ownership of all revisions and denormalized references for editable fields
+    const editableFieldNames = editableCollectionsFields[collectionName]
+    if (editableFieldNames?.length) {
+      editableFieldNames.forEach((editableFieldName) => {
+        transferEditableField({documentId: doc._id, targetUserId, collection, fieldName: editableFieldName})
+      })
+    }
+  }
+  documents.forEach((doc) => {
+    
+  })
+}
+
+const transferEditableField = ({documentId, targetUserId, collection, fieldName = "contents"}) => {
+  // Update the denormalized revision on the document
+  editMutation({
+    collection,
+    documentId,
+    set: {[`${fieldName}.userId`]: targetUserId},
+    unset: {},
+    validate: false
+  })
+  // Update the revisions themselves
+  Revisions.update({ documentId, fieldName }, {$set: {userId: targetUserId}}, { multi: true })
+}
+
+const mergeReadStatusForPost = ({sourceUserId, targetUserId, postId}) => {
+  const sourceUserStatus = ReadStatuses.findOne({userId: sourceUserId, postId})
+  const targetUserStatus = ReadStatuses.findOne({userId: targetUserId, postId})
+  const readStatus = !!(sourceUserStatus?.isRead || targetUserStatus?.isRead)
+  const lastUpdated = new Date(sourceUserStatus?.lastUpdated) > new Date(targetUserStatus?.lastUpdated) ? sourceUserStatus?.lastUpdated : targetUserStatus?.lastUpdated
+  if (targetUserStatus) {
+    ReadStatuses.update({_id: targetUserStatus._id}, {$set: {isRead: readStatus, lastUpdated}})
+  } else if (sourceUserStatus) {
+    // eslint-disable-next-line no-unused-vars
+    const {_id, ...sourceUserStatusWithoutId} = sourceUserStatus
+    ReadStatuses.insert({...sourceUserStatusWithoutId, userId: targetUserId})
+  }
+}
+
+Vulcan.mergeAccounts = async (sourceUserId, targetUserId) => {
+  const sourceUser = Users.findOne({_id: sourceUserId})
+  const targetUser = Users.findOne({_id: targetUserId})
+
+  // Transfer posts
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Posts"})
+
+  // Transfer comments
+  await transferCollection({sourceUserId, targetUserId, collectionName: "Comments"})
+
+  // Transfer conversations
+  await Conversations.update({participantIds: sourceUserId}, {$set: {"participantIds.$": targetUserId}}, { multi: true })
```

Comment:
Potentially breaks the invariant that a user isn't in the same conversation twice, which could have weird effects.